packages/expo-notifications/android/src/main/java/expo/modules/notifications/notifications/triggers/NotificationTriggers.kt

```diff
@@ -0,0 +1,413 @@
+package expo.modules.notifications.notifications.triggers
+
+import android.os.Parcel
+import android.os.Parcelable
+import expo.modules.notifications.notifications.interfaces.NotificationTrigger
+import expo.modules.notifications.notifications.interfaces.SchedulableNotificationTrigger
+import java.io.Serializable
+import java.util.Calendar
+import java.util.Date
+
+open class ChannelAwareTrigger : NotificationTrigger, Serializable {
+  private var mChannelId: String?
+
+  constructor(channelId: String?) {
+    mChannelId = channelId
+  }
+
+  constructor(`in`: Parcel) {
+    mChannelId = `in`.readString()
+  }
+
+  override fun describeContents(): Int {
+    return 0
+  }
+
+  override fun writeToParcel(dest: Parcel, flags: Int) {
+    dest.writeString(mChannelId)
+  }
+
+  override fun getNotificationChannel(): String? {
+    return mChannelId
+  }
+
+  companion object {
+    @JvmField
+    val CREATOR: Parcelable.Creator<ChannelAwareTrigger?> =
+      object : Parcelable.Creator<ChannelAwareTrigger?> {
+        override fun createFromParcel(`in`: Parcel): ChannelAwareTrigger {
+          return ChannelAwareTrigger(`in`)
+        }
+
+        override fun newArray(size: Int): Array<ChannelAwareTrigger?> {
+          return arrayOfNulls(size)
+        }
+      }
+  }
+}
+
+/**
+ * A schedulable trigger representing a notification to be scheduled once per day.
+ */
+class DailyTrigger : ChannelAwareTrigger, SchedulableNotificationTrigger {
+  var hour: Int
+    private set
+  var minute: Int
+    private set
+
+  constructor(hour: Int, minute: Int, channelId: String?) : super(channelId) {
+    this.hour = hour
+    this.minute = minute
+  }
```

Comment:
That should be a [primary class constructor](https://kotlinlang.org/docs/classes.html#constructors).

Reply:
Fixed